pkg/clustermanager/kubetest2/gke.go

```diff
@@ -128,69 +126,45 @@ func Run(opts *Options, cc *GKEClusterConfig) error {
 		kubetest2Flags = append(kubetest2Flags, "--project="+cc.GCPProjectID)
 	}
 
-	return createGKEClusterWithRetries(kubetest2Flags, opts, cc)
-}
+	if cc.PrivateClusterAccessLevel != "" {
+		kubetest2Flags = append(kubetest2Flags, "--private-cluster-access-level="+cc.PrivateClusterAccessLevel)
+		kubetest2Flags = append(kubetest2Flags, "--private-cluster-master-ip-range="+strings.Join(cc.PrivateClusterMasterIPSubnetRange, ","))
+	}
+
+	if opts.TestCommand != "" {
+		kubetest2Flags = append(kubetest2Flags, "--test=exec", "--")
+		kubetest2Flags = append(kubetest2Flags, strings.Split(opts.TestCommand, " ")...)
+	}
 
-func createGKEClusterWithRetries(kubetest2Flags []string, opts *Options, cc *GKEClusterConfig) error {
-	var err error
 	regions := append([]string{cc.Region}, cc.BackupRegions...)
-	for i, region := range regions {
-		flags := append(kubetest2Flags, "--region="+region)
-		if cc.PrivateClusterAccessLevel != "" {
-			flags = append(flags, "--private-cluster-access-level="+cc.PrivateClusterAccessLevel)
-			masterIPRange := fmt.Sprintf("%s.%d/%s", cc.PrivateClusterMasterIPSubnetRange, i, cc.PrivateClusterMasterIPSubnetMask)
-			flags = append(flags, "--private-cluster-master-ip-range="+masterIPRange)
-		}
+	kubetest2Flags = append(kubetest2Flags, "--region="+strings.Join(regions, ","))
+	kubetest2Flags = append(kubetest2Flags, "--retryable-error-patterns='"+retryableErrorPatterns+"'")
 
-		if opts.TestCommand != "" {
-			flags = append(flags, "--test=exec", "--")
-			flags = append(flags, strings.Split(opts.TestCommand, " ")...)
-		}
+	log.Printf("Running kubetest2 with flags: %q", kubetest2Flags)
 
-		log.Printf("Running kubetest2 with flags: %q", flags)
-		command := exec.Command("kubetest2", flags...)
-		var out string
-		out, err = runWithOutput(command)
-		if err != nil {
-			if isRetryableCreationError(out) {
-				log.Print("Cluster creation fails due to unpredictable reasons, will retry creating with different args")
-				continue
-			} else {
-				return err
-			}
-		} else {
-			// Only save the metadata if it's in CI environment and meta data is asked to be saved.
-			if prow.IsCI() && opts.SaveMetaData {
-				saveMetaData(cc, region)
-			}
-			break
-		}
+	command := exec.Command("kubetest2", kubetest2Flags...)
+	command.Stdout = os.Stdout
+	command.Stderr = os.Stderr
+	if err := command.Run(); err != nil {
+		return err
 	}
 
-	return err
-}
-
-// isRetryableCreationError determines if cluster creation should be retried based on the error message.
-func isRetryableCreationError(errMsg string) bool {
-	for _, regx := range retryableCreationErrors {
-		if regx.MatchString(errMsg) {
-			return true
-		}
+	// Only save the metadata if it's in CI environment and meta data is asked to be saved.
+	if prow.IsCI() && opts.SaveMetaData {
+		return saveMetaData(cc, strings.Join(regions, ","))
 	}
-	return false
+	return nil
```

Comment:
My golang is not great, but isn't not having any return statement the same as returning nil?

Reply:
It requires returning explicit `nil`. Named return values can make this simpler, but it's not very commonly used - https://go.dev/tour/basics/7